### ISSUE TITLE
compiler version query can result misleading full version number

### ISSUE THREAD

**Issue by fgvanzee** [flame/blis#262]
`configure` contains a function, `get_compiler_version()`, which uses `-dumpversion` to query the version number if the `cc_vendor` is determined to be `icc`, `gcc`, or `clang`. However, on Ubuntu 18.04, version 7.3.0 of `gcc` returns only `7` when given `-dumpversion`. This leads to confusing output based on the way the version string is parsed:
```
configure: detected Linux kernel version 4.15.0-34-generic.
configure: C compiler search list is: gcc clang cc.
configure: using 'gcc' C compiler.
...
configure: found gcc version 7 (maj: 7, min: 7, rev: 7).
configure: checking for blacklisted configurations due to gcc 7.
```
`configure` reports that 7.7.7 is found, when really it is merely mis-parsing a single line, single char `7` return value.

Switching to `-dumpfullversion` returns the expected `7.3.0`. However, the question is, do `icc` and `clang` both support `-dumpfullversion` (for reasonable ranges of versions)? This is a good question for @devinamatthews .

**Comment by devinamatthews** [comment]
I always use `--version` so I don't know.

**Comment by devinamatthews** [comment]
Also, note that while the system compiler on macOS is clang, Apple uses their own snowflake versioning scheme.

**Comment by fgvanzee** [comment]
@devinamatthews Hmm, I asked because I figured I learned about `-dumpversion` from you. I guess I stumbled upon it myself.

**Comment by fgvanzee** [comment]
Fixed in 743a1a6. Closing.
